File: StartUpExample/StartUpExampleJava/app/src/main/java/com/tinlone/startupexamplejava/initializers/CacheInitializer.java
package com.tinlone.startupexamplejava.initializers;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.startup.Initializer;

import com.tinlone.startupexamplejava.dependencies.Cache;
import com.tinlone.startupexamplejava.dependencies.database.DatabaseProxy;

import java.util.Collections;
import java.util.List;

public class CacheInitializer implements Initializer<Cache> {
    @NonNull
    @Override
    public Cache create(@NonNull Context context) {
        Log.i("loglog", "create: CacheInitializer");
        Cache.initialize(DatabaseProxy.getInstance());
        return Cache.getInstance();
    }

    @NonNull
    @Override
    public List<Class<? extends Initializer<?>>> dependencies() {
        return Collections.singletonList(DatabaseProxyInitializer.class);
    }
}

File: StartUpExample/StartUpExampleKotlin/app/src/main/java/com/tinlone/startupexamplekotlin/initializers/CacheInitializer.kt
package com.tinlone.startupexamplekotlin.initializers

import android.content.Context
import android.util.Log
import androidx.startup.Initializer
import com.tinlone.startupexamplekotlin.dependencies.Cache
import com.tinlone.startupexamplekotlin.dependencies.database.DatabaseProxy

class CacheInitializer : Initializer<Cache> {
    override fun create(context: Context): Cache {
        Log.i("loglog", "create: CacheInitializer")
        Cache.initialize(DatabaseProxy.getInstance())
        return Cache.getInstance()
    }

    override fun dependencies(): List<Class<out Initializer<*>?>> {
        return listOf(DatabaseProxyInitializer::class.java)
    }
}